testdata/default/repros/Issue7457.t.sol

```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity 0.8.18;
+
+import "ds-test/test.sol";
+import "cheats/Vm.sol";
+
+interface ITarget {
+    event AnonymousEventEmpty() anonymous;
+    event AnonymousEventNonIndexed(uint256 a) anonymous;
+
+    event DifferentAnonymousEventEmpty() anonymous;
+    event DifferentAnonymousEventNonIndexed(string a) anonymous;
+
+    event AnonymousEventWith1Topic(uint256 indexed a, uint256 b) anonymous;
+    event AnonymousEventWith2Topics(uint256 indexed a, uint256 indexed b, uint256 c) anonymous;
+    event AnonymousEventWith3Topics(uint256 indexed a, uint256 indexed b, uint256 indexed c, uint256 d) anonymous;
+    event AnonymousEventWith4Topics(
+        uint256 indexed a, uint256 indexed b, uint256 indexed c, uint256 indexed d, uint256 e
+    ) anonymous;
+}
+
+contract Target is ITarget {
+    function emitAnonymousEventEmpty() external {
+        emit AnonymousEventEmpty();
+    }
+
+    function emitAnonymousEventNonIndexed(uint256 a) external {
+        emit AnonymousEventNonIndexed(a);
+    }
+
+    function emitAnonymousEventWith1Topic(uint256 a, uint256 b) external {
+        emit AnonymousEventWith1Topic(a, b);
+    }
+
+    function emitAnonymousEventWith2Topics(uint256 a, uint256 b, uint256 c) external {
+        emit AnonymousEventWith2Topics(a, b, c);
+    }
+
+    function emitAnonymousEventWith3Topics(uint256 a, uint256 b, uint256 c, uint256 d) external {
+        emit AnonymousEventWith3Topics(a, b, c, d);
+    }
+
+    function emitAnonymousEventWith4Topics(uint256 a, uint256 b, uint256 c, uint256 d, uint256 e) external {
+        emit AnonymousEventWith4Topics(a, b, c, d, e);
+    }
+}
+
+// https://github.com/foundry-rs/foundry/issues/7457
+contract Issue7457Test is DSTest, ITarget {
+    Vm constant vm = Vm(HEVM_ADDRESS);
+
+    Target public target;
+
+    function setUp() external {
+        target = new Target();
+    }
+
+    function testEmitEvent() public {
+        vm.expectEmit(false, false, false, true);
+        emit AnonymousEventEmpty();
+        target.emitAnonymousEventEmpty();
+    }
+
+    function testEmitEventNonIndexed() public {
+        vm.expectEmit(false, false, false, true);
+        emit AnonymousEventNonIndexed(1);
+        target.emitAnonymousEventNonIndexed(1);
+    }
+
+    // function testFailEmitDifferentEvent() public {
+    //     vm.expectEmit(false, false, false, true);
+    //     emit DifferentAnonymousEventEmpty();
+    //     target.emitAnonymousEventEmpty();
+    // }
```

Comment:
Note, non-matching events shouldn't be matched but because we don't have an event signature this passes with the proposed implementation - not ideal.

Reply:
This seems like a massive footgun I wouldn't necessarily feel good about introducing—would like your input here @mds1

Reply:
Is the footgun that we comment out this test because we can't make it fail? If so, it makes sense you can never make this fail—both ` event AnonymousEventEmpty() anonymous` and `event DifferentAnonymousEventEmpty() anonymous` have no topics nor data and therefore nothing is emitted as part of this LOG0. So in practice these (and more broadly any pair of anonymous events with the same sig) are indistinguishable